src/store/getters/state-getters.ts

```diff
@@ -1029,3 +1060,13 @@ export const getSpecialResolutionConfirmValid = (state: StateIF): boolean => {
 export const getRestoration = (state: StateIF): RestorationStateIF => {
   return state.stateModel.restoration
 }
+/** Returns true when users can change the sole proprietor (SP).
+ * Restricts the ability of non-staff users from changing the sole
+ * proprietor when the SP is a non-individual.  This restriction has been
+ * added to ensure that changes made in business-edit-ui are also updated by
+ * staff in COLIN */
+export const showSoleProprietorChangeButton = (state: StateIF, getters): boolean => {
+  const isProprietor = getters.getOrgPeople?.[0].roles?.[0]?.roleType === RoleTypes.PROPRIETOR
```

Comment:
Is this a better syntax then `getOrgPeople(state)`?

I think you don't need the extra space between the conditional object and the array index -- I think this works too: `getOrgPeople[0]?.roles[0]?.roleType`.

Reply:
With regards to your first question, yes!  I think it's always better to write pure functions where all dependencies are injected.  Testing functions where the dependencies are not injected is more difficult and refactoring is more difficult.

With regards to your second point, good to know, I'll fix it.